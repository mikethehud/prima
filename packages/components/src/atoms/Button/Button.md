#### Normal

```jsx
<Button text="Hello world" />{' '}
<Button text="Hello world" color="primary" />{' '}
<Button text="Hello world" color="secondary" />{' '}
<Button text="Hello world" color="success" />{' '}
<Button text="Hello world" color="attention" />
```

#### Outline

```jsx
<Button text="Hello world" type="outline" />
<Button text="Hello world" type="outline" color="primary" />
<Button text="Hello world" type="outline" color="secondary" />
<Button text="Hello world" type="outline" color="success" />
<Button text="Hello world" type="outline" color="attention" />
```

#### Ghost

```jsx
<Button text="Hello world" type="ghost" />
<Button text="Hello world" type="ghost" color="primary" />
<Button text="Hello world" type="ghost" color="secondary" />
<Button text="Hello world" type="ghost" color="success" />
<Button text="Hello world" type="ghost" color="attention" />
```

#### Arrow

A button can have an arrow.

```jsx
<Button text="Hello world" color="primary" arrow="true" />
<Button text="Hello world" color="primary" arrow="true" type="outline" />
<Button text="Hello world" color="primary" arrow="true" type="ghost" />
```
