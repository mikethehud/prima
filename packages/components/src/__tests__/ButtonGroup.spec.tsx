/* tslint:disable:jsx-no-lambda */

import { mount, shallow } from "enzyme"
import * as React from "react"
import Button from "../Button/Button"
import ButtonGroup from "../ButtonGroup/ButtonGroup"

describe("ButtonGroup", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ButtonGroup />)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders with one button inside", () => {
    const wrapper = mount(
      <ButtonGroup>
        <Button text="Hello" />
      </ButtonGroup>
    )
    expect(wrapper.find(".button").length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders with multiple buttons inside", () => {
    const wrapper = mount(
      <ButtonGroup>
        <Button text="Hello" />
        <Button text="Hello" />
      </ButtonGroup>
    )
    expect(wrapper.find(".button").length).toEqual(2)
    expect(wrapper).toMatchSnapshot()
  })
})
