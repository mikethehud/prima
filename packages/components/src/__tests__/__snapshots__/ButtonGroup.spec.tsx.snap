// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders correctly 1`] = `
<div
  className="buttonGroup"
/>
`;

exports[`ButtonGroup renders with multiple buttons inside 1`] = `
<ButtonGroup>
  <div
    className="buttonGroup"
  >
    <Button
      arrow={false}
      color="default"
      text="Hello"
      type="normal"
    >
      <button
        className="button normal default"
      >
        <span>
          Hello
        </span>
      </button>
    </Button>
    <Button
      arrow={false}
      color="default"
      text="Hello"
      type="normal"
    >
      <button
        className="button normal default"
      >
        <span>
          Hello
        </span>
      </button>
    </Button>
  </div>
</ButtonGroup>
`;

exports[`ButtonGroup renders with one button inside 1`] = `
<ButtonGroup>
  <div
    className="buttonGroup"
  >
    <Button
      arrow={false}
      color="default"
      text="Hello"
      type="normal"
    >
      <button
        className="button normal default"
      >
        <span>
          Hello
        </span>
      </button>
    </Button>
  </div>
</ButtonGroup>
`;
